packages/platform-browser/src/hydration.ts

```diff
@@ -159,31 +74,36 @@ function provideZoneJsCompatibilityDetector(): Provider[] {
  * export class AppModule {}
  * ```
  *
- * @see {@link withNoDomReuse}
- * @see {@link withNoHttpTransferCache}
+ * `domReuse: false` disables DOM nodes reuse during hydration. Effectively makes
+ * Angular re-render an application from scratch on the client.
+ *
+ * When this option is enabled, make sure that the initial navigation
+ * option is configured for the Router as `enabledBlocking` by using the
+ * `withEnabledBlockingInitialNavigation` in the `provideRouter` call:
  *
- * @param features Optional features to configure additional router behaviors.
+ * This would ensure that the application is rerendered after all async
+ * operations in the Router (such as lazy-loading of components,
+ * waiting for async guards and resolvers) are completed to avoid
+ * clearing the DOM on the client too soon, thus causing content flicker.
+ *
+ *
+ *
+ * @param options to enable domReuse (enabled by default) and specify httpTransferCache options
  * @returns A set of providers to enable hydration.
  *
  * @publicApi
  * @developerPreview
  */
-export function provideClientHydration(...features: HydrationFeature<HydrationFeatureKind>[]):
-    EnvironmentProviders {
-  const providers: Provider[] = [];
-  const featuresKind = new Set<HydrationFeatureKind>();
-
-  for (const {ɵproviders, ɵkind} of features) {
-    featuresKind.add(ɵkind);
-
-    if (ɵproviders.length) {
-      providers.push(ɵproviders);
-    }
-  }
+export function provideClientHydration(options?: {
+  domReuse?: boolean,
+  httpTransferCache?: boolean|HttpTransferCacheOptions&{includePostRequests?: boolean},
```

Comment:
See comments above for context. I think we should include `includePostRequests` into the `HttpTransferCacheOptions` type and use a different type for per-request config.
```suggestion
  httpTransferCache?: boolean|HttpTransferCacheOptions,
```